packages/cactus-core-api/src/main/typescript/plugin/ledger-connector/i-socket-api-client.ts

```diff
@@ -21,5 +21,5 @@ export interface ISocketApiClient<BlockType> {
 
   watchBlocksV1?(
```

Comment:
@outSH Wouldn't this be a breaking change though?

I'm thinking it breaks (crashes) call sites at runtime (if we switch to returning a promise) that look like this:

```typescript
const anObservable = socketApiClient.watchBlocksV1({ ... });
anObservalbe.subscribe(...);
```

Maybe we can add a brand new method instead called `watchBlocksAsyncV1`? That way it's an addition, not a breaking change of the existing API that was locked down when we tagged `1.0.0`.

That would also have the advantage that `watchBlocksAsyncV1` does not need to return `Observable<BlockType> | Promise<Observable<BlockType>>` it can return specifically `Promise<Observable<BlockType>>` and this way the callers will have one less thing to worry about.

Reply:
@petermetz Well, I did not consider this to be used by any client code, only as apiClient contract and inside verifier API. Nevertheless, it is exported as public so yeah, should be treated like public API I suppose. Also, I like the idea of separate naming, this can prevent possible bugs along the way. I've added `watchBlocksAsyncV1` method and adjusted Verifier accordingly.

Reply:
@outSH Yeah, that's the thing, if it's public then someone will probably end up using it somehow even if it wasn't meant to be. Thanks very much for the update!